lib/assert.js

```diff
@@ -210,13 +210,30 @@ function strictDeepEqual(actual, expected) {
     if (!areSimilarTypedArrays(actual, expected)) {
       return false;
     }
-
     // Buffer.compare returns true, so actual.length === expected.length
     // if they both only contain numeric keys, we don't need to exam further
     if (Object.keys(actual).length === actual.length &&
         Object.keys(expected).length === expected.length) {
       return true;
     }
+  } else if (typeof actual.valueOf === 'function') {
```

Comment:
IIUC this also affects objects with custom `valueOf`, so the documentation needs to be updated as well..

Reply:
You are right! I am going to add that to the documentation.

Reply:
Addressed